src/hotspot/os/linux/cgroupSubsystem_linux.cpp

```diff
@@ -123,6 +123,28 @@ CgroupSubsystem* CgroupSubsystemFactory::create() {
   return new CgroupV1Subsystem(cpuset, cpu, cpuacct, pids, memory);
 }
 
+void CgroupSubsystemFactory::set_controller_mount_path(CgroupInfo* cg_infos,
+                                                       int controller,
+                                                       char* name,
+                                                       char* mount_path) {
+  if (cg_infos[controller]._mount_path != NULL) {
+    // On some systems duplicate controllers get mounted in addition to
+    // the main cgroup controllers most likely under /sys/fs/cgroup. In that
+    // case pick the one under /sys/fs/cgroup and discard others.
+    if (strstr(cg_infos[controller]._mount_path, "/sys/fs/cgroup") != cg_infos[controller]._mount_path) {
+      log_warning(os, container)("Duplicate %s controllers detected. Picking %s, skipping %s.",
+                                 name, mount_path, cg_infos[controller]._mount_path);
+      os::free(cg_infos[controller]._mount_path);
+      cg_infos[controller]._mount_path = os::strdup(mount_path);
+    } else {
+      log_warning(os, container)("Duplicate %s controllers detected. Picking %s, skipping %s.",
+                                 name, cg_infos[controller]._mount_path, mount_path);
```

Comment:
Please demote this to `log_debug` over `log_warning`.

Reply:
done